CMakeLists.txt

```diff
@@ -12,53 +10,66 @@ target_include_directories(clap-core INTERFACE include)
 
 install(DIRECTORY include DESTINATION "." OPTIONAL EXCLUDE_FROM_ALL)
 
-add_executable(clap-compile-c EXCLUDE_FROM_ALL src/main.c)
+get_directory_property(parent_dir PARENT_DIRECTORY)
+if ("${parent_dir}" STREQUAL "")
+    set(CLAP_IS_TOPLEVEL 1)
+else ()
+    set(CLAP_IS_TOPLEVEL 0)
+endif ()
```

Comment:
I'll strongly advice against doing it this way. I've seen other projects do similar things (the VST3 SDK is also notorious for it), and it causes a ton of issues and makes things difficult to work with. Instead, I'd just gate this behind a define (e.g. `CLAP_COMPILE_TESTS`), and then pass that option to the GitHub Actions CI with `-DCLAP_COMPILE_TESTS=1`. That's a lot more resilient and doesn't rely on filesystem locations and CMake's internals.

Reply:
Great happy to make that change.